src/datasources.js

```diff
@@ -1,3 +1,45 @@
+/**
+ * Asynchronously loads data sources from vellum.opts().core.dataSources
+ * Currently only supports fixtures
+ *
+ * Format in opts:
+ * dataSources: [
+ *     {
+ *          key: string : data source name (fixture, case, products...)
+ *          endpoint: function or string (URL)
+ *     }
+ * ]
+ *
+ * The endpoint function (or URL) should return the following format:
+ * [
+ *      {
+ *          sourceUri: string (used in the instance definition)
+ *          defaultId: string (used in instance definition)
+ *          intialQuery: string (used in nodeset)
+ *          name: string (text in the dropdown)
+ *          structure: nested dictionary of attributes
+ *          {
+ *              attribute1: {
+ *                  structure: {
+ *                      inner-attribute1: {
+ *                          no_option: boolean (does not show up in the dropdown)
+ *                      }
+ *                  }
+ *              }
+ *          }
+ *      }
+ * ]
```

Comment:
Down here below the example, some further documentation would be nice:
- The value of an element may contain `structure: {...}` describing nested elements and/or attributes.

Also, can we drop the `no_option` thing and instead use the absence of a `structure: {...}` key/value pair to mean the same as `no_option: true`?